backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java

```diff
@@ -118,11 +126,36 @@ private void grantPermissionToTestingUser(Cluster privilegedCluster, String keys
 
     @SuppressWarnings("resource")
     public DockerCassandra() {
-        this("cassandra_3_11_3", AdditionalDockerFileStep.IDENTITY);
+        this("cassandra_3_11_3-"+ UUID.randomUUID().toString(), AdditionalDockerFileStep.IDENTITY);
```

Comment:
`UUID.randomUUID().toString()` this means that we might end up building a new image for each invocation.

An idea to allow multiple builds to hare the same docket daemon could be to realy on a build ID environment variable as a differenciator.

Reply:
I guess you don't feel the pain of trying to build james on a CI for months. 
If this PR goes green, we'll definitely want to merge it and try to optimize later.
You then will be able to try optimizations on the Apache CI build infrastructure and share the result with us.
We don't really like having a 10hours build ... but not having any build at all if far worst.

Reply:
> I guess you don't feel the pain of trying to build james on a CI for months.

That's why I approved straight away :-)

> If this PR goes green, we'll definitely want to merge it and try to optimize later.

I perfectly understand that and agree with it.

Reply:
I have a few ideas of things to try but indeed I would really like to stop maintaining this extremely long lived branch :(

There are 2 reasons to use unique names: 
- it allows for concurrent runs on the same machine (which is a possible cause for [docker cache errors](https://github.com/apache/james-project/pull/264#issuecomment-762826374) ) 
- it makes it easier to diagnose other docker cache related failures. Our working hypothesis here is that there may be a garbage collection process on apache CI which removes images that are not currently in use. it might be a bit too aggressive and collect images that were just build right before we start them. We added detailed logging of docker events in 28d0de15 to diagnose this further

There are a few more ideas for optimization one of the things I intend to try is running in [dockerfile](https://www.jenkins.io/doc/book/pipeline/docker/#dockerfile) to be closer to how linagora's build works 

I also want to merge the [COW fix](https://github.com/apache/james-project/pull/282) and revert "unstable" in a specific branch to see how it impacts both build time and stability

last but not least I hope to find some time in the future to investigate the root cause for the `OutOfMemoryError : Direct buffer memory` which forced me to drop surefire fork reuse. I did some preliminary research and it is most likely a memory leak of off-heap memory. The leak is "resolved" by dropping forks but likely indicates a deeper issue probably with incorrectly sized buffers. Once the leak is resolved we should be able to reactivate reuseFork whih would massively speed up the build (it suceeded once with reuseFork and took around 3h)